src/AdvancedHMC.jl

```diff
@@ -139,9 +143,22 @@ include("sampler.jl")
 export sample
 
 include("abstractmcmc.jl")
-export DifferentiableDensityModel
 
-include("contrib/ad.jl")
+Hamiltonian(metric::AbstractMetric, ℓ::LogDensityModel) = Hamiltonian(
+    metric,
+    Base.Fix1(LogDensityProblems.logdensity, ℓ.logdensity),
+    Base.Fix1(LogDensityProblems.logdensity_and_gradient, ℓ.logdensity),
+)
+function Hamiltonian(metric::AbstractMetric, ℓπ::LogDensityModel, kind::Union{Symbol,Val})
+    ℓ = LogDensityModel(LogDensityProblemsAD.ADgradient(kind, ℓπ.logdensity))
+    return Hamiltonian(metric, ℓ)
+end
```

Comment:
Should this be removed? Or error if the density already has a gradient defined (can be checked with capabilities IIRC)? Otherwise we might override an analytical gradient.

Reply:
Hmm, though I agree with you, if the user actively calls `Hamiltonian(mertic, model, kind)` then the currently expected behavior is that you use `kind` for the gradient computation. That is, the user shouldn't really be calling `Hamiltonian(metric, model, kind)` if they don't want to actually use `kind` for the gradient computation, no? 

And IMO there are def cases where you might want to override a manual impl of the gradient, e.g. if you want to check your own impl vs. "truth" (AD).

Reply:
Seems reasonable.